src/opnsense/mvc/app/models/OPNsense/Base/FieldTypes/BaseField.php

```diff
@@ -481,13 +518,9 @@ public function isRequired()
      * check if this field is unused and required
      * @return bool
      */
-    public function isEmptyAndRequired()
+    public function isEmptyAndRequired(): bool
     {
-        if ($this->internalIsRequired && ($this->internalValue == "" || $this->internalValue == null)) {
-            return true;
-        } else {
-            return false;
-        }
+        return $this->internalIsRequired && $this->isEmpty();
```

Comment:
this will be a bit tricky, when we're going to overwrite `isEmpty` in the boolean field, this will lead to migration issues as the empty intended in this scope is actually "not used" or not provided.

Reply:
I don't really get what you mean. maybe you are referring to code that doesn't exist yet... I can revert that part if you prefer :shrug:

Reply:
hum, should be related to `0`and `false` beeing considered by the `empty()`.
Reverted the test :heavy_check_mark:

Reply:
No problem, I just realized I might have explained it the wrong way around anyway, by example:

We are moving the following block into the base field type, which was the starting point of this journey:

https://github.com/opnsense/core/blob/ec23ffc0ef65ee3fe1f25d23c9690b709edb384c/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/BaseConstraint.php#L41-L47

Now, since "0" is also empty when the type isn't a boolean, we would prevent a migration from saving `0` when the default was `0`, due to:  

https://github.com/opnsense/core/blob/ec23ffc0ef65ee3fe1f25d23c9690b709edb384c/src/opnsense/mvc/app/models/OPNsense/Base/BaseModelMigration.php#L49-L51

The problem here is that "empty" in the constraint has a different meaning than empty in this case, which is also used to make sure unfilled fields are being processed.

Hope this explains the issue a bit, reverting this part of the change for this reason would indeed be better.

Reply:
Done :heavy_check_mark: 
![image](https://github.com/user-attachments/assets/a067dfd7-e156-43aa-9844-b734577ec76b)